Google.Cloud.AspNetCore/Google.AspNetCore.DataProtection.Kms/KmsXmlEncryptor.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright 2019 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Cloud.Kms.V1;
+using Google.Protobuf;
+using Microsoft.AspNetCore.DataProtection.XmlEncryption;
+using System.IO;
+using System.Xml.Linq;
+
+namespace Google.AspNetCore.DataProtection.Kms
+{
+    internal sealed class KmsXmlEncryptor : IXmlEncryptor
+    {
+        private readonly KeyManagementServiceClient _kmsClient;
+        private readonly CryptoKeyName _keyName;
+        private readonly CryptoKeyPathName _keyPathName;
+
+        // TODO: Accept a CryptoKeyPathName as well or instead? (That would allow a specific version to be used for encryption.)
```

Comment:
Because you think we don't need the ability to encrypt with a specific version? It's not clear to me what the benefits would be, but I think it's worth having the TODO to investigate it. (I'd probably ask the KMS team.)